src/pages/task/ActionButtons.tsx

```diff
@@ -259,7 +261,7 @@ export const ActionButtons: React.FC<Props> = ({
     <>
       <PageButtonRow>
         {isBeta() && <PreviousCommits taskId={taskId} />}
-        {isBeta() && (
+        {isBeta() && !isExecutionTask && (
```

Comment:
nit. Maybe PR title should be `Remove history buttons on execution tasks`, since they aren't being disabled but are being entirely removed